jdaviz/configs/default/plugins/gaussian_smooth/tests/test_gaussian_smooth.py

```diff
@@ -157,3 +158,28 @@ def test_spectrum1d_smooth(specviz_helper, spectrum1d):
                                         {'event': 'mousemove', 'domain': {'x': 5500, 'y': 120}})
     assert label_mouseover.as_text() == ('', '', '')
     assert label_mouseover.icon == ''
+
+
+def test_spectrum2d_smooth(specviz2d_helper, spectrum2d):
+    data_label = 'test'
+    dc = specviz2d_helper.app.data_collection
+    specviz2d_helper.load_data(spectrum_2d=spectrum2d, spectrum_2d_label=data_label)
+
+    gs_plugin = specviz2d_helper.plugins['Gaussian Smooth']
+
+    # The Autocollapsed spectrum is given the label of "Spectrum 1D by default"
+    smooth_source_dataset = "Spectrum 1D"
+    gs_plugin.dataset = smooth_source_dataset
+    test_stddev_level = 100.0
+    gs_plugin.stddev = test_stddev_level
+    gs_plugin.smooth()
+
+    assert len(dc) == 3
+    assert dc[2].label == f'{smooth_source_dataset} smooth stddev-{test_stddev_level}'
+
+    # Ensure all marks were created properly (i.e. not in their initialized state)
+    # [0,1] is the default (initialization) value for the marks
+    marks = specviz2d_helper.app.get_viewer('spectrum-viewer').native_marks
+    for mark in marks:
+        assert not np.array_equal(mark.x, np.array([0, 1]))
+        assert not np.array_equal(mark.y, np.array([0, 1]))
```

Comment:
There are only 10 data points. I feel safer if you test specifically for the 10 expected values.

Also, `mark.x` is the spectral axis, right? It should follow the input spectral axis?

Might also want to use `np.testing.assert_allclose` instead of `assert np.<something>`.

Reply:
Good point! Implemented in https://github.com/spacetelescope/jdaviz/pull/2023/commits/6179460e4107d851f48e5c31ab6161f90d624a67.

I couldn't do it for `mark.y` because the marks are different. I'm testing it in a loop rather than each element directly because I'm trying to avoid some javascript error I got when accessing `marks[n]` that didn't show up when I called `for mark in marks`. Strange...